benchmark/tools/compare.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+import sys
+import json
+import argparse
+import math
+import pandas as pd
+
+parser = argparse.ArgumentParser(description="Compare to Ginkgo benchmark outputs")
+parser.add_argument(
+    "--outliers", action="store_true", help="List outliers from the results"
+)
+parser.add_argument(
+    "--outlier-threshold",
+    type=float,
+    default=10,
+    help="At what percentage of deviation (above or below) should outliers be reported",
+)
+parser.add_argument(
+    "--outlier-count",
+    type=int,
+    default=1000,
+    help="How many outliers should be reported per benchmark",
+)
+parser.add_argument("--output", choices=["json", "csv", "markdown"], default="json")
+parser.add_argument("baseline")
+parser.add_argument("comparison")
+args = parser.parse_args()
+keys = {"stencil", "size", "filename", "n", "r", "k", "m"}
```

Comment:
here I am relying on an unchanging set providing consistent ordering, but that's maybe stretching the standard requirements too much. I wanted to avoid relying on external packages as much as possible, but I guess with `pandas` and `tabulate`, that ship has already sailed ;)

Reply:
another possibility would be to use sorted `tuple((k,v), ...)` for the keys